aa_stripe/models.py

```diff
@@ -58,6 +59,89 @@ class Meta:
         ordering = ["id"]
 
 
+class StripeCoupon(StripeBasicModel):
+    DURATION_FOREVER = "forever"
+    DURATION_ONCE = "once"
+    DURATION_REPEATING = "repeating"
+    DURATION_CHOICES = (
+        (DURATION_FOREVER, DURATION_FOREVER),
+        (DURATION_ONCE, DURATION_ONCE),
+        (DURATION_REPEATING, DURATION_REPEATING)
+    )
+
+    coupon_id = models.CharField(max_length=255, help_text=_("Identifier for the coupon"))
+    amount_off = models.PositiveIntegerField(
+        blank=True, null=True, help_text=_("Amount (in the currency specified) that will be taken off the subtotal of"
+                                           "any invoices for this customer."))
+    currency = models.CharField(
+        max_length=3, default="usd", help_text=_("If amount_off has been set, the three-letter ISO code for the"
```

Comment:
I'd try to do choice here if possible. In case coupon is created on our end. Can you just copy the select options from stripe web page source?